Multi currency prices

artursmet opened the pull request:
Fixes #1

patrys commented on django_prices_openexchangerates/models.py:
```diff
@@ -0,0 +1,157 @@
+from decimal import Decimal, ROUND_HALF_UP
```
Missing import of unicode literals.

patrys commented on django_prices_openexchangerates/models.py:
```diff
@@ -0,0 +1,157 @@
+
+    @property
+    def in_base_currency(self):
+        return self.price
```
Please rename `self.price` to `self.base_price` and get rid of this property.

patrys commented on django_prices_openexchangerates/models.py:
```diff
@@ -0,0 +1,157 @@
+            self.base_currency, self.rate, self.to_currency)
+
+
+class MultiCurrencyPrice(object):
```
Can we drop this entirely in favour of:

```python
exchange_currency(Price(10, currency='USD'), 'GBP')
```

patrys commented on django_prices_openexchangerates/models.py:
```diff
@@ -0,0 +1,157 @@
+        return self.price
+
+
+class MultiCurrencyPriceField(PriceField):
```
This should not be needed. A template filter should solve the same use case.

patrys commented on django_prices_openexchangerates/tasks.py:
```diff
@@ -0,0 +1,46 @@
+    for conversion_rate in conversion_rates:
+        new_exchange_rate = exchange_rates[conversion_rate.to_currency]
+        conversion_rate.rate = new_exchange_rate
+        conversion_rate.save()
```
List fields to avoid overwriting all fields?

artursmet commented:
@patrys As we discussed - now there is only one API, `exchange_currency()` method and a few template tags. Please review again.

patrys commented on django_prices_openexchangerates/models.py:
```diff
@@ -0,0 +1,79 @@
+
+
+@python_2_unicode_compatible
+class ConversionRate(models.Model):
```
Please add a `__repr__` that includes the PK (and possibly currency and rate if that's not too much noise).

patrys commented on django_prices_openexchangerates/templatetags/currency_conversion.py:
```diff
@@ -0,0 +1,25 @@
+
+
+@register.simple_tag
+def gross(price, currency):
```
`gross_in_currency`?

patrys commented on django_prices_openexchangerates/templatetags/currency_conversion.py:
```diff
@@ -0,0 +1,25 @@
+
+
+@register.simple_tag
+def net(price, currency):
```
Ditto.

patrys commented on django_prices_openexchangerates/templatetags/currency_conversion.py:
```diff
@@ -0,0 +1,25 @@
+
+
+@register.simple_tag
+def tax(price, currency):
```
And ditto.

patrys commented on django_prices_openexchangerates/tasks.py:
```diff
@@ -0,0 +1,46 @@
+    raise ImproperlyConfigured('OPENEXCHANGERATES_API_KEY is required')
+
+
+class ExchangeRates(object):
```
As this only has one method, wouldn't it be more readable as a function?

```python
def extract_rate(rates, currency):
    base_rate = rates[settings.DEFAULT_CURRENCY]
    return rates[currency] / base_rate
```

patrys commented on django_prices_openexchangerates/templatetags/currency_conversion.py:
```diff
@@ -0,0 +1,25 @@
+
+from ..utils import exchange_currency
+
+
```
Superfluous newline

patrys commented on django_prices_openexchangerates/utils.py:
```diff
@@ -0,0 +1,51 @@
+            raise ValueError('conversion rate for %s is 0', rate.to_currency)
+
+    return Price(currency=to_currency, net=converted_price,
+                 history=price.history)
```
It would be nice to be able to see the currency switch in the history somehow.

artursmet commented:
@patrys I added `PriceModifier` and I convert currencies there. I also refactored old `exchange_currency` function, now it's more readable.
Example with price modifier:

```python
In [3]: exchange_currency(Price(10, currency='PLN'), 'EUR').history
Out[3]: ((Price('10', currency='PLN') * CurrencyConversion(base_currency=PLN, to_currency=USD, rate=0.2670)) * CurrencyConversion(base_currency=USD, to_currency=EUR, rate=0.8840))
```
